frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -106,7 +106,12 @@ export const eventListener: TelemetryEventListener = async (
       }
       break;
     case 'page':
-      (window as any).analytics.page(undefined, properties, anonymousIP);
+      {
+        const titleElement = document.querySelector('title');
+        const titleValue = titleElement.getAttribute('data-telemetry');
+        properties.title = titleValue || document.title;
+        (window as any).analytics.page(undefined, properties, anonymousIP);
+      }
```

Comment:
Instead of applying the title property just for segment, we should do that when the 'page' event is created.

Two times in  `public/components/app.jsx`

Reply:
In Segment, we are checking the event type and based the event type we are making network call. I added above code for eventType is page. Basically all the page events will be captured here.

Reply:
But other telemetry listeners might be also interested in the same page title. This should be a general telemetry improvement, not just for the segment implementation.

Reply:
Added in app.jsx